plugins/webhooks/webhooks.js

```diff
@@ -303,54 +293,58 @@ function pushEvent(request, reply) {
  *  - Opening a PR should sync the pipeline (creating the job) and start the new PR job
  *  - Syncing a PR should stop the existing PR job and start a new one
  *  - Closing a PR should stop the PR job and sync the pipeline (disabling the job)
- * @method github
  * @param  {Hapi.Server}    server
  * @param  {Object}         options
- * @param  {String}         options.secret    GitHub Webhook secret to sign payloads with
+ * @param  {String}         [options.secret]    GitHub Webhook secret to sign payloads with
  * @param  {Function}       next
  */
 module.exports = (server, options) => {
-    // Register the hook interface
-    server.register(githubWebhooks);
-    // Add the auth strategy
-    server.auth.strategy('githubwebhook', 'githubwebhook', {
-        secret: options.secret
-    });
-
-    // Now use it
-    return {
+    const scm = server.root.app.pipelineFactory.scm;
+    const obj = {
         method: 'POST',
-        path: '/webhooks/github',
+        path: '/webhooks',
         config: {
-            description: 'Handle events from GitHub',
+            description: 'Handle webhook events',
             notes: 'Acts on pull request, pushes, comments, etc.',
-            tags: ['api', 'github', 'webhook'],
-            auth: {
-                strategies: ['githubwebhook'],
-                payload: 'required'
-            },
-            handler: (request, reply) => {
-                const eventType = request.headers['x-github-event'];
-                const eventId = request.headers['x-github-delivery'];
-
-                request.log(['webhook-github', eventId], `Received event ${eventType}`);
-
-                switch (eventType) {
-                case 'ping':
-                    return reply().code(204);
-                case 'pull_request':
-                    return pullRequestEvent(request, reply);
-                case 'push':
-                    return pushEvent(request, reply);
-                default:
-                    return reply(boom.badRequest(`Event ${eventType} not supported`));
-                }
-            }
+            tags: ['api', 'webhook'],
+            handler: (request, reply) =>
+                scm.parseHook(request.headers, request.payload)
+                .then((parsed) => {
+                    const eventType = parsed.type;
+                    const eventId = parsed.hookId;
+
+                    request.log(['webhook', eventId], `Received event type ${eventType}`);
+
+                    switch (eventType) {
+                    case 'pr':
+                        return pullRequestEvent(request, reply, parsed);
+                    case 'repo':
+                        return pushEvent(request, reply, parsed);
+                    default: // for the event 'ping'
+                        return reply().code(204);
+                    }
+                })
+                .catch(err => reply(boom.wrap(err)))
         }
     };
+
+    if (Object.keys(scm).includes('github')) {
```

Comment:
Please don't have if statements hard coded for plugin implementations.  Defeats the purpose of plugins.  This needs to either be moved to GitHub SCM or the implementation replicated in there (validates the signature).